### PR TITLE
Edits to JOSS paper

### PR DESCRIPTION
**Pull request by kyleniemeyer**
Edits to the JOSS paper for the Thermal Hydraulics Module: https://github.com/openjournals/joss-reviews/issues/6146

### CODE REVIEW

File: joss_paper.bib [hunk 1]
```diff
@@ -14,16 +14,15 @@ @article{lindsay2022moose
...
+     doi = {10.1016/j.softx.2022.101202},
 keywords = {Multiphysics, Object-oriented, Finite-element, Framework},
 }
 
 @techreport{relap7theory,
 title = {{RELAP-7} Theory Manual},
 author = {R. A. Berry and L. Zou and H. Zhao and H. Zhang and J. W. Peterson and R. C. Martineau and S. Y. Kadioglu and D. Andrs},
 institution = {Idaho National Laboratory},
-number = {INL/EXT-14-31366},
+howpublished = {INL/EXT-14-31366},
```
Comment (joshuahansel) [inline joss_paper.bib:25]: I think `number` is the appropriate field - why change?

File: joss_paper.bib [hunk 2]
```diff
@@ -65,21 +64,24 @@ @article{slaughter2021continuous
...
+  doi={10.1080/00295450.2020.1838877}
 }
 
 @techreport{sqa,
-  title={Software Quality Assurance Plan for MOOSE and MOOSE-Based Applications (PLN-4005)},
+  title={Software Quality Assurance Plan for {MOOSE} and {MOOSE}-Based Applications ({PLN}-4005)},
   institution={Idaho National Laboratory},
   year={2020},
-  note = {Revision 7}
+  note = {Revision 7},
```
Comment (joshuahansel) [inline joss_paper.bib:84]: I don't think a comma should go after the last item, but maybe this is a moot point if this reference needs to be removed anyway (I'm currently asking around if there's a public SQA plan reference).

**Review by joshuahansel — CHANGES_REQUESTED** [review]

File: joss_paper.bib (hunk 1 above)
Comment (kyleniemeyer) [reply joss_paper.bib:25]: ah, I thought that our citation formatting didn't recognize `number` for `techreport` type, but at second glance I can see that it did show up property. More importantly, we should add the DOI 10.2172/1262488

File: joss_paper.bib (hunk 2 above)
Comment (kyleniemeyer) [reply joss_paper.bib:84]: This is just force of habit on my part—commas after the final list entry are totally acceptable, but yes unnecessary. (Same is true in most programming language lists, dicts, etc.).

**Comment by kyleniemeyer** [comment]
@joshuahansel I updated the RELAP-7 theory manual citation, to use `number` again and also updated to the 2018 version, with appropriate DOI: https://www.osti.gov/biblio/1498781

**Comment by kyleniemeyer** [comment]
Thanks @GiudGiud - I just amended the last commit to include that reference and force-pushed it.

**Comment by joshuahansel** [comment]
> Looks like we wont allow civet to run on PRs to branches that are not next @joshuahansel we may want to change that

I think this is fine, just a little awkward: this PR into my feature branch is untested (fine), and then I'll do a PR into next from this branch, which you can review and is tested.

**Comment by joshuahansel** [comment]
@kyleniemeyer I checked on the SQA reference. There is no public reference for the plan itself, but the first reference in that section (slaughter2021continuous) does discuss the plan a bit, so I propose that we delete the reference and the corresponding entry in the .bib file. If you approve, then either you can apply those changes, or we merge this one and then I do it afterwards.

**Comment by kyleniemeyer** [comment]
@joshuahansel ok, I went ahead and replaced that citation with the existing one (and updated the bibliographic data for the latter—realized that it still had placeholder info).

**Review by joshuahansel — APPROVED** [review]

**Pull request merged**